src/components/textarea/textarea.ts

```diff
@@ -146,15 +147,21 @@ export default class SlTextarea extends LitElement {
 
   /** Selects all the text in the textarea. */
   select() {
-    return this.input.select();
+    this.input.select();
   }
 
   /** Gets or sets the textarea's scroll position. */
-  scrollPosition(position?: { top?: number; left?: number }) {
-    if (position) {
-      if (typeof position.top === 'number') this.input.scrollTop = position.top;
-      if (typeof position.left === 'number') this.input.scrollLeft = position.left;
-      return;
+  scrollPosition(position?: { top?: number; left?: number }): void;
+  scrollPosition(): { top: number; left: number };
+  scrollPosition(position?: { top?: number; left?: number }): { top: number; left: number } | undefined {
```

Comment:
These are TypeScript overload declarations. So this says that if you pass an object then the return value is void because it's setting a value, but if you don't pass any arguments then it returns an object because it is getting a value. And then the last line is the actual function declaration which has the combination of both.

I actually do see an issue here I should change `position?:` to `position:` in the first line.

The benefit of this is that the caller will get the appropriate TS return value inferred based on the arguments passed.